### ISSUE TITLE
Symlink build behavior change

### ISSUE THREAD

**Issue by fzerorubigd** [constabulary/gb#528]
Before #511 this structure was ok : 
```
.
├── src
│   └── somecode -> ../vendor/someplace/somecode
└── vendor
    └── someplace
        └── somecode
            └── main.go
```
this code : https://github.com/fzerorubigd/gb511, 
`gb build` dose not build anything with error `FATAL: command "build" failed: no packages supplied
`, but `gb build somecode` build it ok. 
but in new version I get this error : 
```
$ gb build someplace                                        
FATAL: command "build" failed: failed to resolve import path "someplace": import "someplace": not a directory
```

I don't know if this is a bug or not, but its somehow change in the old behavior.

**Comment by davecheney** [comment]
@fzerorubigd I don't believe that old behaviour was supported. The go/build package was ruthless in its rejection of symlinks.

The workaround from #100 was to cd into $PROJECT/vendor and build, that would place any binary dependencies in $PROJECT/vendor/bin.


I'll check an older revision and confirm.

Wrt. your example, gb only builds code that is in $PROJECT/src/. If there are other dependencies in $GOROOT/src or $PROJECT/vendor/src then they would be built _transitively_  but were never considered to be the roots of any package built (modulo #505 which is hard to exploit)

**Comment by davecheney** [comment]
Another workaround is to move the vendored `someplace/somecode` into $PROJECT/src.

**Comment by fzerorubigd** [comment]
In my case, I can not move the code inside the src. there is a library attached to this code. splitting that package into two separate package is out of my control. its something like go-bindata (see #100)

**Comment by davecheney** [comment]
@fzerorubigd how about this as a counterproposal

gb build $PKG will build that package if it finds it in $PROJECT/src or
$PROJECT/vendor/src ?

**Comment by fzerorubigd** [comment]
That's great. +1!

**Comment by davecheney** [comment]
@fzerorubigd ok, i'll look into how complicated it is to make that happen. For reference

cd $PROJECT; gb build 

will only build the contents of $PROJECT/src/...

but 

cd $PROJECT; gb build $PKG

will build $PKG if it is found in $PROJECT/src/$PKG or $PROJECT/vendor/src/$PKG

How does that sound ?

**Comment by fzerorubigd** [comment]
This is ok, I already use this by setting the GOPATH and using the Go tool in a make file (as a workaround)

**Comment by davecheney** [comment]
@fzerorubigd closed in #555.

**Comment by fzerorubigd** [comment]
@davecheney Thank you!
